### DISCUSSION TITLE
Serialization warning for symbols as optional?

### DISCUSSION THREAD

**Discussion by key88sf** [sidekiq/sidekiq#5303] [General]
Hi - I noticed that the latest version of Sidekiq now gives a warning when passing symbols as arguments to the async job, since they aren't technically a JSON primitive (as noted in https://github.com/mperham/sidekiq/wiki/Best-Practices).

However, I'm wondering if we can make an exception for symbols, since in many Ruby/Rails apps they are used interchangeably with strings, and I believe the general intent with symbols is that developers expect them to be natively converted to strings?

Perhaps an initializer setting can be used to allow symbols to still be passed as arguments without raising warnings/errors, and developers would just need to be responsible for knowing they will arrive as strings (not symbols) when the job is invoked? I 100% agree that complex objects should raise an error and never be passed, but I think symbols do not fall into this bucket and should be allowed if developers understand the behavior.

WDYT?

**Comment by mperham** [comment]
Please read the changelog to learn how to disable the warnings.

https://github.com/mperham/sidekiq/blob/main/Changes.md#640

**Reply by key88sf** [reply]
@mperham I don't want to disable the warnings for everything though. Just for symbols.

**Reply by mperham** [reply]
I'm not going to make exceptions. Entirely JSON safe or unsafe. No half-way measures.

**Reply by mperham** [reply]
If you are using Symbols as arguments today, update your codebase. I've documented Symbols as unsafe for **5+ years** now.
